### PR TITLE
[stable/nginx-ingress] Add PodSecurityPolicy support

### PR DESCRIPTION
**Pull request by verwilst**
Signed-off-by: Bart Verwilst <bart@verwilst.be>

**What this PR does / why we need it**: Adds initial PodSecurityPolicy support

### CODE REVIEW

File: stable/nginx-ingress/templates/podsecuritypolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{{- if .Values.rbac.pspEnabled }}
+apiVersion: extensions/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: {{ template "nginx-ingress.fullname" . }}
```
Comment (cpanato) [inline stable/nginx-ingress/templates/podsecuritypolicy.yaml]: Need to add the labels to match the chart helm practices.

Comment (verwilst) [reply stable/nginx-ingress/templates/podsecuritypolicy.yaml]: Fixed

File: stable/nginx-ingress/templates/podsecuritypolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    rule: 'MustRunAs'
+    ranges:
+      # Forbid adding the root group.
+      - min: 1
+        max: 65535
+  readOnlyRootFilesystem: false
+  seLinux:
+    rule: 'RunAsAny'
+  hostPorts:
+    -
```
Comment (davidkarlsen) [inline stable/nginx-ingress/templates/podsecuritypolicy.yaml]: should max be moved up one line?

File: stable/nginx-ingress/values.yaml [hunk 3]
```diff
@@ -354,6 +354,10 @@ defaultBackend:
 rbac:
   create: true
 
+  # If true, create & use Pod Security Policy resources
+  # https://kubernetes.io/docs/concepts/policy/pod-security-policy/
+  pspEnabled: false
```
Comment (davidkarlsen) [inline stable/nginx-ingress/values.yaml]: would it make sense to have the ci-test values to true for psp? so that we get it tested?

Comment (verwilst) [reply stable/nginx-ingress/values.yaml]: Not sure if psp is enabled during ci-test?

File: stable/nginx-ingress/templates/podsecuritypolicy.yaml (hunk 2 above)
Comment (verwilst) [reply stable/nginx-ingress/templates/podsecuritypolicy.yaml]: Fixed

File: stable/nginx-ingress/values.yaml (hunk 3 above)
Comment (cpanato) [reply stable/nginx-ingress/values.yaml]: i learned something new today, you can follow this: https://github.com/helm/charts/tree/master/test#providing-custom-test-values

Comment (paulczar) [inline stable/nginx-ingress/values.yaml]: This should be added to the README.   I'm not sure I agree that it should belong under `rbac` in the values, I'd like it to be `podSecurityPolicy.enabled: false`

**Review by paulczar — CHANGES_REQUESTED** [review]

File: stable/nginx-ingress/templates/podsecuritypolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{{- if .Values.rbac.pspEnabled }}
```
Comment (davidkarlsen) [inline stable/nginx-ingress/templates/podsecuritypolicy.yaml]: mismatch with readme

File: stable/nginx-ingress/templates/role.yaml [hunk 5]
```diff
@@ -79,4 +79,11 @@ rules:
     verbs:
       - create
       - patch
+{{- if .Values.rbac.pspEnabled }}
```
Comment (davidkarlsen) [inline stable/nginx-ingress/templates/role.yaml]: same here

File: stable/nginx-ingress/values.yaml (hunk 3 above)
Comment (verwilst) [reply stable/nginx-ingress/values.yaml]: I mean, I'm not sure that the cluster used in testing needs to have podsecuritypolicies enabled for it to have effect. I can add it and see what happens.

File: stable/nginx-ingress/values.yaml [hunk 6]
```diff
@@ -354,6 +354,11 @@ defaultBackend:
 rbac:
   create: true
 
+# If true, create & use Pod Security Policy resources
+# https://kubernetes.io/docs/concepts/policy/pod-security-policy/
+podSecurityPolicy:
+  enabled: false
```
Comment (davidkarlsen) [inline stable/nginx-ingress/values.yaml:360]: would it be good to configure the ci-tests to run with this enabled to get test-coverage?
See https://github.com/helm/charts/tree/master/test

Comment (verwilst) [reply stable/nginx-ingress/values.yaml:360]: Already committed. Let's see what gives.

Comment (davidkarlsen) [reply stable/nginx-ingress/values.yaml:360]: seems it worked just fine!

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**